app/queries/get_document.rb

```diff
@@ -0,0 +1,25 @@
+module Queries
+  module GetDocument
+    def self.call(content_id, locale = nil)
+      locale_to_use = locale || ContentItem::DEFAULT_LOCALE
+
+      begin
+        retries ||= 0
+        Document.transaction(requires_new: true) do
+          Document.find_or_create_by!(
+            content_id: content_id,
+            locale: locale_to_use
+          ).lock!
```

Comment:
I reckon we can refactor this to not have a second query for the lock. Will have a think.